File: service-two-product-listing/adapters/adapters.py
import json

def parseRequest(request):
    """Returns a dict with the X-User-Id header value.

    Args:
        arg (request): HTTP Get Request

    Returns:
        dict: Dict containing the userId

    """
    parsedRequest = {
        "userId": request.headers.get("X-User-Id")
    }
    return parsedRequest

File: service-one-individual-discount/adapters/adapters.js
/**
 * Parse a IndividualDiscount Request.
 * @param {IndividualDiscountRequest} request 
 * @returns {Object} With productId and userId from the request.
 */
function parseRequest(request) {

    var productId = request.productId;
    var userId = request.userId;
    return({productId: productId, userId: userId});
    
}

module.exports = {
    parseRequest: parseRequest
}